ui/app/pages/settings/networks-tab/networks-tab.component.js

```diff
@@ -208,49 +200,61 @@ export default class NetworksTab extends PureComponent {
                 rpcUrl={rpcUrl}
                 chainId={chainId}
                 ticker={ticker}
-                onClear={() => {
+                onClear={(shouldUpdateHistory = true) => {
                   setNetworksTabAddMode(false)
-                  setSelectedSettingsRpcUrl(null)
+                  setSelectedSettingsRpcUrl('')
+                  if (shouldUpdateHistory && !isFullScreen) {
+                    history.push(NETWORKS_ROUTE)
+                  }
```

Comment:
This spares us from re-pushing `NETWORKS_ROUTE` unnecessarily. Used in `NetworkForm.componentWillUnmount`.